### ISSUE TITLE
glyph disapear with mosh

### ISSUE THREAD

**Issue by jerome-diver** [romkatv/powerlevel10k#1598]
# What about ```mosh``` ?
when i connect to a remote machine with ```mosh```, glyph icons dispear from the shell prompt.
When i connect with ```ssh``` icons glyph are printed correctly.

# look:
## with ```mosh``` (no glyph of BSD on left)
![mosh access glyph failed](https://user-images.githubusercontent.com/11176132/136889820-02951991-2b5f-480b-8d96-b7188281a254.png)
## with ```ssh``` (BSD glyph has been printed)
![ssh icon glyph ok](https://user-images.githubusercontent.com/11176132/136889827-2aabc27b-c653-4b50-9066-7ca5aef81e79.png)

# what are my systems ?
## locale
Fedora-34 and mosh-1.3.2, with ZSH shell and oh-my-zsh + powerlevel10k
## remote
FreBSD-12.2 and mosh-1.3.2 with ZSH shell and oh-my-zsh + powerlevel10k

# Is it really in relation ?
From ```mosh``` github repos [issues 1157](https://github.com/mobile-shell/mosh/issues/1157) post, it seems to be in relation with this: [#351](https://github.com/mobile-shell/mosh/issues/351)... but is it correct and then, how to fix the problem ?

# Question
How to fix that for i can get glyph printed with ```mosh``` connected on my FreeBSD-12.2 remote VPS system ?

**Comment by jerome-diver** [comment]
thank you @romkatv. I'm going right now to link this issue on the one open in github mosh side (and still hope it will not be an old ping-pong ugly game beween each side that users are the big looser).

**Comment by romkatv** [comment]
There is no need to link to this issue. The problem is not related to powerlevel10k. Please use the `printf` command to describe the issue to mosh devs.

**Comment by romkatv** [comment]
> which other soiurce code can call the glyph char if not the "(old ?) unicode unassigned code" ?

I don't understand the question. Can you rephrase?

**Comment by romkatv** [comment]
> > which other soiurce code can call the glyph char if not the "(old ?) unicode unassigned code" ?
> 
> I don't understand the question. Can you rephrase?

I think I know what you mean. The FreeBSD logo that you see in your prompt is U+F30C. This is from *Private Use Area*. It's not unassigned. The same goes for U+E0B0 (the triangle).

**Comment by jerome-diver** [comment]
@romkatv yes, i wrote wrong command, yours is working fine in:
- my locale shell
- my remote 'Archlinux' machine shell (even through mosh connection)

__but not in FreeBSD remote mosh connected shell machine (and only this one).__ 

That is why i think it is strange that it works with mosh in Archlinux but onot with FreeBSD (it should have a logic link i miss)

to rephrase the question, is there an other one source code to call to print the glyph char than this '\UF30C' (line 449 of the internal/icons.zsh file) ?

**Comment by romkatv** [comment]
> but not in FreeBSD remote mosh connected shell machine (and only this one).

Then go ahead and bring it up with FreeBSD and/or mosh devs.

> to rephrase the question, is there an other one source code to call to print the glyph char than this '\UF30C' (line 449 of the internal/icons.zsh file) ?

Are you asking how to print `\UF30C`?

```zsh
printf '\uF30C\n'
```

If this is not what you are asking, please clarify.

> and to be honest, i'm not really understand myself what is the real problem...

The mosh devs are telling you that `wcwidth` on your FreeBSD box is broken. I don't know if this is actually true. This can be verified by writing a simple C program that invokes `wcwidth` on U+E0B0.

**Comment by jerome-diver** [comment]
ok, thank you for your help, i understand a bit better and i'm going to catch any freebsd "folks" and mosh guys together. Thank you again for your time.

**Comment by jerome-diver** [comment]
It is aa freebsd problem that will be fixed from version 12.3 (or actuall 12.2 SSTABLE release, who is NOT the current one) the 21 october 2021 OR on version 13.0

It is in relation with the wcwidth() lc_ctype set that should be updated.
